File: Scanner/CommunicationsFramework/Applications/Production/DataPublicationsEnvironment/DPEProxy/Server/WindowsApplication/Forms/passwordConfirmationForm.vb
Namespace GeneralPurposeForms

    Public Class passwordConfirmationForm

        Private _passwordToVerify As String

        Public Sub New(ByVal passwordToVerify As String)

            ' This call is required by the Windows Form Designer.
            InitializeComponent()

            ' Add any initialization after the InitializeComponent() call.
            Me._passwordToVerify = passwordToVerify
        End Sub


        Private Sub btnOk_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOk.Click
            Try
                If Me.txtPassword.Text.Length <= 0 Then
                    Throw New Exception("No password specified")
                End If
                If Me.txtPassword.Text = Me._passwordToVerify Then
                    Me.DialogResult = Windows.Forms.DialogResult.OK
                Else
                    Me.DialogResult = Windows.Forms.DialogResult.Cancel
                End If
            Catch ex As Exception
                MsgBox(ex.Message)
            End Try
        End Sub

        Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
            Try
                Me.DialogResult = Windows.Forms.DialogResult.Cancel
            Catch ex As Exception
            End Try
        End Sub

        Private Sub txtPassword_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles txtPassword.KeyDown
            Try
                If e.KeyCode = Windows.Forms.Keys.Enter Then
                    Me.btnOk_Click(Nothing, Nothing)
                End If
            Catch ex As Exception

            End Try
        End Sub

      
        Private Sub passwordConfirmationForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Try
                Me.txtPassword.Focus()
            Catch ex As Exception
            End Try
        End Sub


    End Class


End Namespace

File: Scanner/CommunicationsFramework/Applications/Production/DataPublicationsEnvironment/DPEServer/Server/DPEServerApp/Forms/passwordConfirmationForm.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports


namespace DPEServerApp
{
	namespace GeneralPurposeForms
	{
		
		public partial class passwordConfirmationForm
		{
			
			private string _passwordToVerify;
			
			public passwordConfirmationForm(string passwordToVerify)
			{
				
				// This call is required by the Windows Form Designer.
				InitializeComponent();
				
				// Add any initialization after the InitializeComponent() call.
				this._passwordToVerify = passwordToVerify;
			}
			
			
			public void btnOk_Click(System.Object sender, System.EventArgs e)
			{
				try
				{
					if (this.txtPassword.Text.Length <= 0)
					{
						throw (new Exception("No password specified"));
					}
					if (this.txtPassword.Text == this._passwordToVerify)
					{
						this.DialogResult = System.Windows.Forms.DialogResult.OK;
					}
					else
					{
						this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
					}
				}
				catch (Exception ex)
				{
					MessageBox.Show(ex.Message);
				}
			}
			
			public void btnCancel_Click(System.Object sender, System.EventArgs e)
			{
				try
				{
					this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
				}
				catch (Exception)
				{
				}
			}
			
			public void txtPassword_KeyDown(object sender, System.Windows.Forms.KeyEventArgs e)
			{
				try
				{
					if (e.KeyCode == System.Windows.Forms.Keys.Enter)
					{
						this.btnOk_Click(null, null);
					}
				}
				catch (Exception)
				{
					
				}
			}
			
			
			public void passwordConfirmationForm_Load(System.Object sender, System.EventArgs e)
			{
				try
				{
					this.txtPassword.Focus();
				}
				catch (Exception)
				{
				}
			}
			
			
		}
		
		
	}
	
	
}
